boa/src/builtins/number/mod.rs

```diff
@@ -555,6 +555,187 @@ impl Number {
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-isfinite-number
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isFinite
+    pub(crate) fn global_is_finite(
+        _this: &Value,
+        args: &[Value],
+        ctx: &mut Interpreter,
+    ) -> ResultValue {
+        if let Some(val) = args.get(0) {
+            let number = ctx.to_number(val)?;
+            Ok(number.is_finite().into())
+        } else {
+            Ok(false.into())
+        }
+    }
+
+    /// Builtin javascript 'isNaN(number)' function.
+    ///
+    /// Converts the argument to a number, throwing a type error if the conversion is invalid.
+    ///
+    /// If the number is NaN true is returned.
+    ///
+    /// Otherwise false is returned.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-isnan-number
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN
+    pub(crate) fn global_is_nan(
+        _this: &Value,
+        args: &[Value],
+        ctx: &mut Interpreter,
+    ) -> ResultValue {
+        if let Some(val) = args.get(0) {
+            let number = ctx.to_number(val)?;
+            Ok(number.is_nan().into())
+        } else {
+            Ok(true.into())
+        }
+    }
+
+    /// `Number.isFinite( number )`
+    ///
+    /// Checks if the argument is a number, returning false if it isn't.
+    ///
+    /// If the number is NaN, +∞, or -∞ false is returned.
+    ///
+    /// Otherwise true is returned.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-number.isfinite
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isFinite
+    pub(crate) fn number_is_finite(
+        _this: &Value,
+        args: &[Value],
+        _ctx: &mut Interpreter,
+    ) -> ResultValue {
+        Ok(Value::from(if let Some(val) = args.get(0) {
+            match val {
+                Value::Integer(_) => true,
+                Value::Rational(number) => number.is_finite(),
+                _ => false,
+            }
+        } else {
+            false
+        }))
+    }
+
+    /// `Number.isInteger( number )`
+    ///
+    /// Checks if the argument is an integer.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-number.isinteger
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger
+    pub(crate) fn number_is_integer(
+        _this: &Value,
+        args: &[Value],
+        _ctx: &mut Interpreter,
+    ) -> ResultValue {
+        Ok(args.get(0).map_or(false, Self::is_integer).into())
```

Comment:
```suggestion
                Ok(Value::from(match args.get(0) {
            Some(Value::Integer(_)) => true,
            Some(Value::Rational(number)) => Self::is_integer(number), 
            _ => false,
        }))
```
Its better if we do it this way this gives us a fast path for when the value is a `Value::Integer(_)` since it is always an integer because it is stored as a `Value::Integer(i32)`.

Reply:
Doesn't is_integer have a fast path for Value::Integer(_) already? Doing this would save a function call as is_integer() might not be inlined, but without that they should already be the same.

Reply:
> Doesn't is_integer have a fast path for Value::Integer(_) already? Doing this would save a function call as is_integer() might not be inlined, but without that they should already be the same.

Ow sorry, I was mistaken :sweat_smile: , 

> Doing this would save a function call as is_integer() might not be inlined

I would suggest putting `#[inline]` on `is_integer` and is `is_float_integer` this will help with inlining, this is just a hint to the compiler.

Reply:
It was already on is_float_integer, but I've done that for is_integer now :)